### DISCUSSION TITLE
How do you get access to the wiki files?

### DISCUSSION THREAD

**Discussion by logangamesdaily** [community/community#68564] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

I would like to display my GitHub repository's wiki on a webpage, but I can't embed/iframe or get the data directly and just display the html via JS. Anyway I could do this?

**Comment by logangamesdaily** [comment]
What I want to do is access the raw wiki files

**Comment by seyLu** [comment, marked as answer]
#### Hi @An-Unnamed-Developer,

Since the wiki is it's own git repository, there's currently no way to programmatically access its contents via the API. 

What you can do though, since the wiki is a separate repo and is in markdown, is host the wiki on Github Pages or any static website hosting platform (eg. Cloudfare, Vercel, etc).

If you chose Github Pages, the github wiki uses a different markup than what Jekyll supports, so you also need to convert it to github flavored markdown. I haven't tested these tools personally, but you're free to try them out yourself:
- [pandoc](https://github.com/jgm/pandoc)  
- [mediawiki-to-gfm](https://github.com/outofcontrol/mediawiki-to-gfm)
- [github-wikito-converter](https://github.com/yakivmospan/github-wikito-converter)

[Or use hugo and a perl script to convert github wiki markup to be compliant with hugo syntax](https://discourse.gohugo.io/t/my-tips-to-turn-a-github-wiki-into-hugo-website/21217).

You can read more on:
- [Github blog: Git-backed wikis](https://github.blog/2010-08-12-making-github-more-open-git-backed-wikis/)
- [StackOverflow: Is there any API endpoint to read Github wiki](https://stackoverflow.com/questions/58334455/is-there-any-api-endpoint-in-git-hub-to-read-wikis)

**Reply by logangamesdaily** [reply]
Actually you can by entering this url, `https://raw.githubusercontent.com/wiki/user/repo/page.md` and this returns the data in the page 😄

**Reply by Nileena** [reply]
Hi, I was able to get content via raw api as mentioned. Is there any way to list the pages in wiki?

**Reply by rcriii42** [reply]
Hi Nileena,

On the wiki on github, at the lower right of the screen, is a link to clone the wiki locally. The wiki pages are in a repository at `https://github.com/user/project.wiki.git`. The wiki pages are in a repository at `https://github.com/user/project.wiki.git`.  It will clone into a directory called `project.wiki` with the pages as markdown files.

**Reply by roke-julian-lockhart** [reply]
> [`#discussioncomment-9937608`](https://github.com/orgs/community/discussions/68564#discussioncomment-9937608)

@An-Unnamed-Developer and @rcriii42, this is elaborated on at [`medium.com/@thecybermonk/the-secret-github-wiki-repo-and-ci-cd-for-docs-fafe2583b72e`](https://medium.com/@thecybermonk/the-secret-github-wiki-repo-and-ci-cd-for-docs-fafe2583b72e#:~:text=GitHub%20wikis%20are%20powered%20by,sjawhar/circleci%2Dorb%2Dgithub%2Dwiki.).

**Comment by janwilmake** [comment]
Hi, I made it work.

https://github.com/Netflix/Hystrix/wiki -> https://wiki.forgithub.com/Netflix/Hystrix/wiki

It will soon also be a part of https://uithub.com so you can browse it as text (good for LLM)
